zed: Fix image mapping in kolla-images.py, add tests and CI

markgoddard opened the pull request:
- **kolla-images.py: Add a check-image-map command**
- **kolla-images.py: Fix image to container exceptions using check-image-map**
- **kolla-images.py: Misc fixes**
- **CI: Add check-image-map and check-hierarchy to check-tags workflow**

markgoddard committed:
kolla-images.py: Add a check-image-map command

This command checks the image mapping against Kolla Ansible variables.

The *_image variables in Kolla Ansible define the mapping between
containers and images. Ensure that the mapping defined in this script
matches the one in Kolla Ansible.

markgoddard committed:
kolla-images.py: Fix image to container exceptions using check-image-map

We recently had an issue where images were built for ironic, but this
did not include dnsmasq due to not matching the regex. The 'ironic' tag
was updated in kolla-image-tags.yml, which matches ironic_dnsmasq
container, but no such dnsmasq image existed. This issue would have been
caught the check-tags workflow with the correct mapping added in this
change.

markgoddard committed:
kolla-images.py: Misc fixes

markgoddard committed:
CI: Add check-image-map and check-hierarchy to check-tags workflow

These commands ensure that the image map and hierarchy defined in
kolla-images.py matches Kolla Ansible.

Also update the path filter to run the check-tags workflow when pulp.yml
or kolla-images.py is modified.

markgoddard committed:
CI: Trigger build-kayobe-image for check-tags job

MoteHue commented on tools/kolla-images.py:
```diff
@@ -39,15 +39,19 @@
     ],
-    "neutron-eswitchd": [
+    "neutron-mlnx-agent": [
         "neutron_mlnx_agent",
```
Think we need to add eswitchd to this list. Both neutron-mlnx-agent and neutron-eswitchd use the neutron-mlnx-agent image. Example from customer site:
```
3701af5de4ca   172.25.2.221:80/stackhpc/neutron-mlnx-agent:2023.1-rocky-9-20240809T090211                  "dumb-init --single-…"   7 days ago    Up 7 days                          neutron_eswitchd
a2b5b6ca72c6   172.25.2.221:80/stackhpc/neutron-mlnx-agent:2023.1-rocky-9-20240809T090211                  "dumb-init --single-…"   7 days ago    Up 7 days                          neutron_mlnx_agent
```

markgoddard replied:
You are right, hawk eye. This wasn't caught because neutron-mlnx-agent isn't in the list of supported images in pulp.yml, although I did see the issue before I filtered out unsupported images.

markgoddard committed:
kolla-images.py: Add image map exception for neutron_eswitchd

This was not caught because neutron-mlnx-agent is not in the list of
supported images in pulp.yml.

MoteHue reviewed (approved):
LGTM now, nice one
